synapse/handlers/register.py

```diff
@@ -386,11 +386,32 @@ async def _create_and_join_rooms(self, user_id: str) -> None:
                 room_alias = RoomAlias.from_string(r)
 
                 if self.hs.hostname != room_alias.domain:
+                    # If the alias is remote, try to join the room. This might fail
+                    # because the room might be invite only, but we don't have any local
+                    # user in the room to invite this one with, so at this point that's
+                    # the best we can do.
                     logger.warning(
-                        "Cannot create room alias %s, "
-                        "it does not match server domain",
+                        "Cannot automatically create room with alias %s as it isn't"
+                        " local, trying to join the room instead",
                         r,
                     )
+
+                    (
+                        room,
+                        remote_room_hosts,
+                    ) = await room_member_handler.lookup_room_alias(room_alias)
```

Comment:
This method raising a `SynapseError` during registration would probably be quite bad. We probably want to wrap it in a `try/except`, and log on failure to join.

Reply:
It's already inside a try/except.